pkg/webhook/instance_admission.go

```diff
@@ -29,99 +34,148 @@ func (ia *InstanceAdmission) Handle(ctx context.Context, req admission.Request)
 	switch req.Operation {
 
 	case v1beta1.Create:
-		// trigger "deploy" for freshly created Instances: req.Object contains the created object
-		new := &kudov1beta1.Instance{}
-		if err := ia.decoder.DecodeRaw(req.Object, new); err != nil {
-			return admission.Errored(http.StatusBadRequest, err)
-		}
-
-		// since we don't yet enforce the existence of the 'deploy' plan in the OV, we check for its existence
-		// and decline Instance creation if the plan is not found
-		ov, err := instance.GetOperatorVersion(new, ia.client)
-		if err != nil {
-			log.Printf("InstanceAdmission: Error getting operatorVersion %s for instance %s/%s: %v", new.Spec.OperatorVersion.Name, new.Namespace, new.Name, err)
-			admission.Errored(http.StatusInternalServerError, err)
-		}
+		return handleCreate(ia, req)
 
-		plan := kudov1beta1.SelectPlan([]string{kudov1beta1.DeployPlanName}, ov)
-		if plan == nil {
-			return admission.Denied(fmt.Sprintf("failed to create an Instance %s/%s: couldn't find '%s' plann in the operatorVersion", new.Namespace, new.Name, kudov1beta1.DeployPlanName))
-		}
+	case v1beta1.Update:
+		// req has both old and new Instance objects
+		return handleUpdate(ia, req)
 
-		new.Spec.PlanExecution.PlanName = *plan
+	default:
 		return admission.Allowed("")
+	}
+}
 
-	case v1beta1.Update:
-		// call validation for Instance Updates
-		old, new := &kudov1beta1.Instance{}, &kudov1beta1.Instance{}
+func handleCreate(ia *InstanceAdmission, req admission.Request) admission.Response {
+	// trigger "deploy" for freshly created Instances: req.Object contains the created object
+	new := &kudov1beta1.Instance{}
+	if err := ia.decoder.DecodeRaw(req.Object, new); err != nil {
+		return admission.Errored(http.StatusBadRequest, err)
+	}
 
-		// req.Object contains the updated object
-		if err := ia.decoder.DecodeRaw(req.Object, new); err != nil {
-			return admission.Errored(http.StatusBadRequest, err)
-		}
+	// if namespace is not set, 'default' is explicitly set, otherwise it is empty (at this point)
```

Comment:
oh this is actually not true I think, I forgot to follow up with this 🙈 Take a look here https://github.com/kubernetes/kubernetes/issues/88282

Reply:
Oh, I completely forgot about this hack. Thank you! 🙇